### PR TITLE
Add Scoreboard components

### LINKED ISSUE #34: Allow attaching components to teams

### PR DESCRIPTION
**Pull request by Pyrofab**
This PR implements scoreboard and team components.
- Scoreboard components are attached to Scoreboard instances (accessible from any World) - they are essentially alternatives to level components. They may replace level components at some point, as they are less reliant on a fragile class hierarchy.
- Team components are attached to Team instances - they can be used to implement shared data between entities in the same team.

Both component providers support synchronized components. They get attached using the [experimental static component system](https://github.com/OnyxStudios/Cardinal-Components-API/wiki/Registering-and-using-a-component#static-registration)

Closes #34

### CODE REVIEW

**Comment by Pyrofab** [comment]
@haykam821 opinions ?

File: cardinal-components-scoreboard/src/main/java/dev/onyxstudios/cca/mixin/scoreboard/MixinScoreboardState.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private Scoreboard scoreboard;
+
+    @Inject(method = "toTag", at = @At("RETURN"))
+    private void saveComponents(CompoundTag tag, CallbackInfoReturnable<CompoundTag> cir) {
+        ((InternalComponentProvider) this.scoreboard).getComponentContainer().toTag(tag);
+    }
+
+    @Inject(method = "fromTag", at = @At("RETURN"))
+    private void loadComponents(CompoundTag tag, CallbackInfo ci) {
+        if (this.scoreboard != null) {  // weird deferred loading thing
```
Comment (UpcraftLP) [inline cardinal-components-scoreboard/src/main/java/dev/onyxstudios/cca/mixin/scoreboard/MixinScoreboardState.java:54]: tiny_potato
[:star:](# ":iapprove:")

File: cardinal-components-scoreboard/src/main/resources/mixins.cardinal_components_scoreboard.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "required": true,
+  "minVersion": "0.7.11",
```
Comment (UpcraftLP) [inline cardinal-components-scoreboard/src/main/resources/mixins.cardinal_components_scoreboard.json]: should probably be 0.8

File: gradle.properties [hunk 3]
```diff
@@ -10,7 +10,7 @@ loader_version=0.8.8+build.202
 fabric_api_version=0.13.1+build.370-1.16
 
 #Publishing
-mod_version = 2.4.1
+mod_version = 2.4.2
```
Comment (UpcraftLP) [inline gradle.properties]: not 2.5?

**Review by UpcraftLP — CHANGES_REQUESTED** [review]

**Review by UpcraftLP — APPROVED** [review]

**Pull request merged**